dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship.unittests/Integration/GetApprenticeshipsAsProviderIntegrationTests.cs

```diff
@@ -0,0 +1,112 @@
+
+        public GetApprenticeshipsAsProviderIntegrationTests()
+        {
+            _telemetryClient = new TelemetryClient();
+            _appCache = new CachingService();
+            _configuration = new Mock<IConfiguration>();
+            _cosmosDbHelper = new Mock<ICosmosDbHelper>();
+            _cosmosSettings = Options.Create(new CosmosDbCollectionSettings());
+
+            _referenceDataResponse = new Mock<Func<HttpRequestMessage, CancellationToken, Task<HttpResponseMessage>>>();
+            _referenceDataService = new ReferenceDataService(new HttpClient(new MockHttpMessageHandler(_referenceDataResponse.Object)) { BaseAddress = new Uri("https://test.com") });
+            _referenceDataServiceClient = new ReferenceDataServiceClient(_telemetryClient, new Mock<IOptions<ReferenceDataServiceSettings>>().Object, _appCache, _referenceDataService);
+            _providerResponse = new Mock<Func<HttpRequestMessage, CancellationToken, Task<HttpResponseMessage>>>();
+            _providerService = new ProviderService(new HttpClient(new MockHttpMessageHandler(_providerResponse.Object)) { BaseAddress = new Uri("https://test.com") });
+            _providerServiceClient = new ProviderServiceClient(new Mock<IOptions<ProviderServiceSettings>>().Object, _appCache, _providerService);
+            
+            _DASHelper = new DASHelper(_telemetryClient, _referenceDataServiceClient);
+            _apprenticeshipService = new ApprenticeshipService(_telemetryClient, _cosmosDbHelper.Object, _cosmosSettings, _providerServiceClient, _DASHelper, _appCache);
+
+            _function = new GetApprenticeshipsAsProvider(_appCache, _apprenticeshipService, _configuration.Object);
+        }
+
+        [Fact]
+        public async Task Run_ReturnsExpectedResult()
+        {
+            _referenceDataResponse.Setup(s => s.Invoke(It.IsAny<HttpRequestMessage>(), It.IsAny<CancellationToken>()))
+                .Returns<HttpRequestMessage, CancellationToken>(async (r, ct) => new HttpResponseMessage(HttpStatusCode.OK)
+                {
+                    Content = new StringContent(await File.ReadAllTextAsync("Integration/fechoices.json"))
+                });
+
+            _providerResponse.Setup(s => s.Invoke(It.IsAny<HttpRequestMessage>(), It.IsAny<CancellationToken>()))
+                .Returns<HttpRequestMessage, CancellationToken>(async (r, ct) => new HttpResponseMessage(HttpStatusCode.OK)
+                {
+                    Content = new StringContent(await File.ReadAllTextAsync("Integration/providers.json"))
+                });
+
+            _cosmosDbHelper.Setup(s => s.GetLiveApprenticeships(It.IsAny<DocumentClient>(), It.IsAny<string>()))
+                .Returns(() => JsonConvert.DeserializeObject<List<Apprenticeship>>(File.ReadAllText("Integration/apprenticeships.json")));
+
+            var request = new Mock<HttpRequest>();
+
+            var result = await _function.Run(request.Object, NullLogger.Instance);
```

Comment:
I personally don't like generically named test subjects, especially in the land of diffs. It requires me to go and re-load extra context when reading the code.
```suggestion
            var result = await _getApprenticeshipsAsProvider.Run(request.Object, NullLogger.Instance);
```

Reply:
But it makes refactoring a breeze :)